## Я взял данные из максимальной производительности из 1 задачи(W=97)

$$
f(x) = x_{12} + x_{13}  + x_{21} + x_{22} + x_{31} + x_{33} \rightarrow max \\  
0.007x_{12} + 0.006x_{13} + 0.004x_{21} + 0.009x_{22} + 0.003x_{31} + 0.005x_{33} \leq 97 \\
9x_{12} + 14x_{13} \leq 79800 \\
16x_{21} + 9x_{22} \leq 49800 \\
12x_{31} + 12x_{33} \leq 83400 \\
x_{12} + x_{13} - x_{21} - x_{22} = 0 \\
x_{12} + x_{13} - x_{31} - x_{33} = 0 \\
x_{ij} \geq 0, i \in {1,2,3}, j \in {1,2,3}  
$$


Таблица:

| *  | x11   | x12   | x13   | x21   | x22   | x23   | x31   | x32 | x33   | Неравенство | b      |
|----|-------|-------|-------|-------|-------|-------|-------|-----|-------|-------------|--------|
| c  | 0     | 1     | 1     | 1     | 1     | 0     | 1     | 0   | 1     | -           | max(-1)|
| y1 | 0     | 0.007 | 0.006 | 0.004 | 0.009 | 0     | 0.003 | 0   | 0.005 | <=          | 97     |
| y2 | 0     | 9     | 14    | 0     | 0     | 0     | 0     | 0   | 0     | <=          | 79800  |
| y3 | 0     | 0     | 0     | 16    | 9     | 0     | 0     | 0   | 0     | <=          | 49800  |
| y4 | 0     | 0     | 0     | 0     | 0     | 0     | 12    | 0   | 12    | <=          | 83400  |
| y5 | 0     | 1     | 1     | -1    | -1    | 0     | 0     | 0   | 0     |  =          | 0      |
| y6 | 0     | 1     | 1     | 0     | 0     | 0     | -1    | 0   | -1    |  =          | 0      |

In [2]:
import numpy as np
from cvxopt import matrix, solvers

c = matrix([0, -1, -1, -1, -1, 0, -1, 0, -1], tc='d') #Целевая функция (минусы, потому что решаем задачу максимизации) 

G = matrix([[0, 0.007,  0.006,  0.004, 0.009, 0, 0.003,  0, 0.005],   # Коэффициенты при ограничениях-неравенствах (вида <=)
            [    0,     9, 14,  0,     0,     0,     0,  0,     0],
            [    0,     0,  0, 16,     9,     0,     0,  0,     0],
            [    0,     0,  0,  0,     0,     0,    12,  0,    12],
            [    0,     1,  1, -1,     -1,    0,     0,  0,     0],
            [    0,     1,  1,  0,     0,     0,    -1,  0,    -1],
            [   -1,     0,  0,  0,     0,     0,     0,  0,     0],
            [    0,    -1,  0,  0,     0,     0,     0,  0,     0],
            [    0,     0, -1,  0,     0,     0,     0,  0,     0],
            [    0,     0,  0, -1,     0,     0,     0,  0,     0],
            [    0,     0,  0,  0,    -1,     0,     0,  0,     0],
            [    0,     0,  0,  0,     0,    -1,     0,  0,     0],
            [    0,     0,  0,  0,     0,     0,    -1,  0,     0],
            [    0,     0,  0,  0,     0,     0,     0, -1,     0],
            [    0,     0,  0,  0,     0,     0,     0,  0,    -1]], tc='d')
h = matrix([97,79800,49800,83400,0,0,0,0,0,0,0,0,0,0,0], tc='d')
solution = solvers.lp(c, G.T, h, solver='glpk')
print('Status:', solution['status'])
print('Objective :', solution['primal objective'])
print('x = \n', solution['x'])


Status: optimal
Objective : -17377.333333333332
x = 
 [ 0.00e+00]
[ 9.09e-13]
[ 5.21e+03]
[ 4.11e+02]
[ 4.80e+03]
[ 0.00e+00]
[ 6.95e+03]
[ 0.00e+00]
[ 0.00e+00]



## Загруженность оборудования

In [24]:
x_11 = 0.00e+00
x_12 = 9.09e-13
x_13 = 5.21e+03
x_21 = 4.11e+02
x_22 = 4.80e+03
x_23 = 0.00e+00
x_31 = 6.95e+03
x_32 = 0.00e+00
x_33 = 0.00e+00
time1 = 9*x_12+14*x_13
time2 = 16*x_21+9*x_22
time3 = 12*x_31+12*x_33
if (time1==79800):
    print ("\n1 группа полностью загружена")
elif (time2==49800):
    print ("\n2 группа полностью загружена")
elif (time3==83400):
    print ("\n3 группа полностью загружена")
time1,time2,time3


3 группа полностью загружена


(72940.00000000001, 49776.0, 83400.0)

In [4]:
# Исследование интервала осуществимости 
dh = matrix([0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);     # приращение к вектору правых частей
#print(h,dh,h+dh)
solution1 = solvers.lp(c, G.T, h + dh, solver='glpk')
print('Status:', solution1['status'])
print('Objective:', -solution1['primal objective'], 'delta:', -solution1['primal objective']-(-solution['primal objective']))
print('x = \n', solution1['x'])
print('z = \n',solution1['z'])

prev_z= -solution['primal objective']
a = 1
while (True):
    solution_i= solvers.lp(c, G.T, h + dh*a, solver='glpk')
    if solution_i['status'] != 'optimal':
        print("Couldn''t find a solution")
        break
    new_z= -solution_i['primal objective']
    delta_z= new_z-prev_z
    print('Increment %d: obj=%.3f delta=%.3f' % (a, new_z, delta_z))
    if abs(delta_z - (0.06)) > 1e-6:
        print("Basis changed at increment %d" % (a,))
        break
    prev_z = new_z
    a = a + 1

Status: optimal
Objective: 17377.4 delta: 0.06666666666933452
x = 
 [ 0.00e+00]
[-9.09e-13]
[ 5.21e+03]
[ 4.11e+02]
[ 4.80e+03]
[ 0.00e+00]
[ 6.95e+03]
[ 0.00e+00]
[ 0.00e+00]

z = 
 [ 9.33e+01]
[-0.00e+00]
[ 6.67e-02]
[ 6.00e-02]
[ 4.40e-01]
[-0.00e+00]
[-0.00e+00]
[ 9.33e-02]
[-0.00e+00]
[-0.00e+00]
[-0.00e+00]
[-0.00e+00]
[-0.00e+00]
[-0.00e+00]
[ 1.87e-01]

Increment 1: obj=17377.400 delta=0.067
Basis changed at increment 1


## В результате было получено следующее: 
“Status: optimal” - План работы оборудования найден. В x (Количество изделений) содержатся значения переменных $x_{ij}$ , а 17377.400 – это максимальное возможное значение целевой функции. 
Таким образом, оптимальный план предпологает:  
$x_{11}$ = 0,  
$x_{12}$ = 0,  
$x_{13}$ = 5210,  
$x_{21}$ = 411,  
$x_{22}$ = 4800,  
$x_{23}$ = 0,  
$x_{31}$ = 6950,  
$x_{32}$ = 0,  
$x_{33}$ = 0
## Анализ загруженности оборудования  
time1 = 72940.0  
time2 = 49776.0  
time3 = 83400.0(3 группа полностью загружена)  
Получили теневые цены (z).“Status: optimal”,говорит о том, что оптимальный план работы оборудования найден.  
## Параметрическое изменение правых частей  
Т.к в решении есть загруженная группа оборудования(3 группа), то средствами параметрического изменения правых частей было исследовано влияние величины фонда рабочего времени 3 группы оборудования на структуру решения (изменение фонда рабочего времени в сторону увеличения и уменьшения).  
Поняли, что на 1 инкременте, значение теневой цены изменилось, т.е структура
решения тоже изменилась.  
Increment 1: obj=17377.400 delta=0.067  
Basis changed at increment 1 
Как видно из вывода, сгенерированного скриптом, без изменения состава базисных переменных данное ограничение не может быть ослаблено, что даст, и увеличение количество производства изделий нет смысла.